File: ProblemSets/2/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>


// Functions prototypes
int formula(float, float);
float char_count(string);
float word_count(string);
float sentence_count(string);

int main(void)
{
    // To get the text from user
    string text = get_string("Text: ");

    // Counting characters, words and sentences
    float char_counted = char_count(text);
    float word_counted = word_count(text);
    float sentence_counted = sentence_count(text);

    // Calculating L and S
    float L = (char_counted * 100) / word_counted;
    float S = (sentence_counted * 100) / word_counted;

    // Calculating the grade
    int grade = formula(L, S);

    // Printing the grade
    // If it was betweeen 1 and 16
    if (grade >= 1 && grade < 16)
    {
        printf("Grade %i\n", grade);
    }

    // If it was grater than 16
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }

    // If it was lesser than 1
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

}

// All the functions

// The formula
int formula(float L, float S)
{
    int result;
    float grade = (0.0588 * L) + (-0.296 * S) - 15.8;
    grade = round(grade);
    result = (int)grade;

    return result;
}

// Character count
float char_count(string text)
{
    int length = strlen(text);
    char ch;
    int count = 0;

    for (int i = 0; i < length; i++)
    {
        ch = tolower(text[i]);

        if (ch >= 'a' && ch <= 'z')
        {
            count += 1;
        }
    }
    return (float)count;
}

// Word count
float word_count(string text)
{
    int length = strlen(text);
    int space_count = 0;

    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ')
        {
            space_count += 1;
        }
    }

    if (space_count == 0)
    {
        return (float)space_count;
    }

    else
    {
        space_count += 1;
        return (float)space_count;
    }
}

// Sentence count
float sentence_count(string text)
{
    int length = strlen(text);
    int count = 0;

    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count += 1;
        }
    }

    return (float)count;
}

File: ProblemSets/6/readability.py
from cs50 import get_string

# First, some functions


def formula(L, S):

    grade = (0.0588 * L) + (-0.296 * S) - 15.8
    grade = round(grade)
    result = int(grade)
    return result


def char_count(text):

    length = len(text)
    count = 0

    for i in range(0, length):
        ch = text[i].lower()

        if ch >= 'a' and ch <= 'z':
            count += 1

    return float(count)


def word_count(text):

    space_count = 0
    length = len(text)

    for i in range(0, length):
        if text[i] == ' ':
            space_count += 1

    if space_count == 0:
        return float(space_count)

    else:
        space_count += 1
        return float(space_count)


def sentence_count(text):

    length = len(text)
    count = 0

    for i in range(0, length):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            count += 1

    return float(count)


# To get the text from user
text = get_string("Text: ")

# Counting characters, words and sentences
char_counted = char_count(text)
word_counted = word_count(text)
sentence_counted = sentence_count(text)

# Calculating L and S
L = (char_counted * 100) / word_counted
S = (sentence_counted * 100) / word_counted

# Calculating the grade
grade = formula(L, S)

# Printing the grade
if grade >= 1 and grade < 16:
    print(f"Grade {grade}")

# If it's more than 16
elif grade >= 16:
    print("Grade 16+")

# If it's less than 1
elif grade < 1:
    print("Before Grade 1")